src/compute/src/sink/persist_sink.rs

```diff
@@ -285,17 +286,37 @@ where
         AsyncOperatorBuilder::new(format!("{} mint_batch_descriptions", operator_name), scope);
 
     let (mut output, output_stream) = mint_op.new_output();
+    let (mut data_output, data_output_stream) = mint_op.new_output();
 
+    // The description input and the data-passthrough input are both driven by this input, so
+    // they use a standard input connection.
     let mut desired_input = mint_op.new_input(desired_stream, Pipeline);
-    let mut persist_feedback_input =
-        mint_op.new_input_connection(persist_feedback_stream, Pipeline, vec![Antichain::new()]);
+
+    let mut persist_feedback_input = mint_op.new_input_connection(
+        persist_feedback_stream,
+        Pipeline,
+        // Neither output's capabilities should depend on the feedback input.
+        vec![Antichain::new(), Antichain::new()],
+    );
 
     let shutdown_button = mint_op.build(move |mut capabilities| async move {
+        // Non-active workers should just pass the data through.
         if !active_worker {
+            capabilities.clear();
```

Comment:
The capabilities are dropped anyway after the while loop. Are we already clearing them here to drop them as soon as possible, before we potentially have to wait for the input? I think this could use an explaining comment, as it's not immediately clear whether we need this for correctness or just for efficiency.

Also is there any difference between `clear`ing the vec and dropping it?

Reply:
Will add a comment!